gensim/models/ensemblelda.py

```diff
@@ -0,0 +1,1268 @@
+
+        Finally, stable topics can be retreived using
+        get_topics().
+
+        Parameters
+        ----------
+        eps : float
+            epsilon for the CBDBSCAN algorithm, having the same
+            meaning as in classic DBSCAN clustering.
+            default: 0.1
+        min_samples : number
+            The minimum number of sampels in the neighborhood
+            of a topic to be considered a core in CBDBSCAN.
+            default: int(self.num_models / 2)
+        min_cores : number
+            how many cores a cluster has to have,
+            to be treated as stable topic. That means, how many topics
+            that look similar have to be present, so that the average
+            topic in those is used as stable topic.
+            default: min(3, max(1, int(self.num_models /4 +1)))
+
+        """
+        # if new models were added to the ensemble, the distance
+        # matrix needs to be generated again
+        if self.asymmetric_distance_matrix_outdated:
+            logger.info("asymmetric distance matrix is outdated due to add_model")
+            self.generate_asymmetric_distance_matrix()
+
+        self.generate_topic_clusters(eps, min_samples)
+        self.generate_stable_topics(min_cores)
+        self.generate_gensim_representation()
+
+    # GENSIM API
+    # to make using the ensemble in place of a gensim model as easy as possible
+
+    def get_topics(self):
+        return self.stable_topics
+
+    def has_gensim_representation(self):
+        """checks if stable topics area vailable and if the internal
+        gensim representation exists. If not, raises errors"""
+        if self.classic_model_representation is None:
+            if len(self.stable_topics) == 0:
+                raise ValueError("no stable topic was detected")
+            else:
+                raise ValueError("use generate_gensim_representation() first")
+
+    def __getitem__(self, i):
+        """see https://radimrehurek.com/gensim/models/ldamodel.html"""
+        self.has_gensim_representation()
+        return self.classic_model_representation[i]
+
+    def inference(self, *posargs, **kwArgs):
+        """see https://radimrehurek.com/gensim/models/ldamodel.html"""
+        self.has_gensim_representation()
+        return self.classic_model_representation.inference(*posargs, **kwArgs)
+
+    def log_perplexity(self, *posargs, **kwArgs):
+        """see https://radimrehurek.com/gensim/models/ldamodel.html"""
+        self.has_gensim_representation()
+        return self.classic_model_representation.log_perplexity(*posargs, **kwArgs)
+
+    def print_topics(self, *posargs, **kwArgs):
+        """see https://radimrehurek.com/gensim/models/ldamodel.html"""
+        self.has_gensim_representation()
+        return self.classic_model_representation.print_topics(*posargs, **kwArgs)
+
+    @property
+    def id2word(self):
+        return self.gensim_kw_args["id2word"]
+
+
+class CBDBSCAN():
```

Comment:
Please add a docstring.

Reply:
Added docstring, as well as full explanation for the CBDBSCAN algorithm (because we haven't published a proper public paper reference yet, I thought that would help).